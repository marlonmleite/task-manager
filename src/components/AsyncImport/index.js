import Loadable from 'react-loadable'
import { LoadingContent } from '../Loading'

const AsyncImport = config => Loadable({
  ...config,
  loading: LoadingContent,
})

export default AsyncImport
