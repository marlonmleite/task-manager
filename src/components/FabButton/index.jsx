import React, { Component } from 'react'
import PlusIcon from 'core/assets/svg/plus.svg'
import FabButtonStyled from './styled'

class FabButton extends Component {

  static defaultProps = {
    color: 'primary',
  }

  render() {
    return (
      <FabButtonStyled {...this.props}>
        <PlusIcon fill="#fff" />
      </FabButtonStyled>
    )
  }

}

export default FabButton
