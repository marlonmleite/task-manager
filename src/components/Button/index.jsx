import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import { Loading } from 'components/Loading'
import { ButtonStyled, LoadingContainer } from './styled'

class Button extends PureComponent {

  static defaultProps = {
    color: 'primary',
    type: 'button',
    onClick: null,
    loading: false,
  }

  static propTypes = {
    children: PropTypes.node.isRequired,
    onClick: PropTypes.func,
    color: PropTypes.string,
    type: PropTypes.string,
    loading: PropTypes.bool,
  }

  render() {
    const { children, ...props } = this.props

    return (
      <ButtonStyled {...props}>
        <span>{children}</span>
        {props.loading && (
          <LoadingContainer>
            <Loading size="14px" color="secondary" />
          </LoadingContainer>
        )}
      </ButtonStyled>
    )
  }

}

export default Button
