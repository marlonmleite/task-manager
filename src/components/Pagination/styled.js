import styled from 'styled-components'
import Button from 'components/Button'

const ButtonMore = styled(Button)`
  font-weight: bold;
  text-transform: uppercase;
  box-shadow: 0 0 4px rgba(0,0,0,.14), 0 4px 8px rgba(0,0,0,.28);
`

export default ButtonMore
