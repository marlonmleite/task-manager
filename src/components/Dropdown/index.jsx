import PropTypes from 'prop-types'
import React, { Component } from 'react'
import Button from '../Button'
import { Dropdown as DropdownStyled, DropdownContent, DropdownButton, DropdownOption, CaretStyled } from './styled'

class Dropdown extends Component {

  state = {
    open: false,
  }

  toggle = (event) => {
    const { open } = this.state
    const nextOpen = !open

    if (event.defaultPrevented) {
      return
    }

    if (nextOpen) {
      window.document.addEventListener('click', this.toggle)
    } else {
      window.document.removeEventListener('click', this.toggle)
    }

    this.setState({ open: nextOpen })
  }


  render() {
    const { open } = this.state
    const { text, children, color, right } = this.props

    return (
      <DropdownStyled color={color}>
        <Button onClick={this.toggle} color={color}>
          <DropdownButton>
            <span>{text}</span>
            <CaretStyled color={color} />
          </DropdownButton>
        </Button>
        {open && (
          <DropdownContent right={right}>
            {children}
          </DropdownContent>
        )}
      </DropdownStyled>
    )
  }

}

Dropdown.defaultProps = {
  color: 'primary',
  right: false,
}

Dropdown.propTypes = {
  text: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  color: PropTypes.string,
  right: PropTypes.bool,
}

export { Dropdown, DropdownOption }
