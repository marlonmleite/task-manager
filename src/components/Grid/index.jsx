import PropTypes from 'prop-types'
import React, { Component } from 'react'
import map from 'lodash/map'
import { Grid as GridStyled, GridItem, Remove, Finish } from './styled'

class Grid extends Component {

  render() {
    const { items, render } = this.props

    return (
      <GridStyled>
        {map(items, (item, index) => {
          const key = `grid-${index}`

          return (
            <GridItem key={key}>
              {render(item, index)}
              <div className="grid-actions">
                <Remove />
                <Finish />
              </div>
            </GridItem>
          )
        })}
      </GridStyled>
    )
  }

}

Grid.propTypes = {
  items: PropTypes.array.isRequired,
  render: PropTypes.func.isRequired,
}

export default Grid
