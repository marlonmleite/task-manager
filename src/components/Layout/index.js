import PropTypes from 'prop-types'
import styled from 'styled-components'

export const Col = styled.div`
  box-sizing: border-box;
  flex: 0 0 auto;
  max-width: 100%;
  padding: 0 15px;

  ${({ size }) => `
    flex-basis: ${size};
    max-width: ${size};
  `}
`

export const Row = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: ${({ justify }) => justify};
`

Col.defaultProps = {
  size: '100%',
}

Col.propTypes = {
  size: PropTypes.string,
}

Row.defaultProps = {
  justify: 'flex-start',
}

Row.propTypes = {
  justify: PropTypes.string,
}
