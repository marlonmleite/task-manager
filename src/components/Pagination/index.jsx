import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import { Row } from 'components/Layout'
import ButtonMore from './styled'

const INCREMENT = 1

class Pagination extends PureComponent {

  static defaultProps = {
    loading: false,
  }

  static propTypes = {
    page: PropTypes.number.isRequired,
    hasMore: PropTypes.bool.isRequired,
    onChange: PropTypes.func.isRequired,
    loading: PropTypes.bool,
  }

  handleClick = () => {
    const { onChange, page, loading } = this.props

    if (!loading) {
      onChange(page + INCREMENT)
    }
  }

  render() {
    const { hasMore, loading } = this.props

    if (!hasMore) {
      return null
    }

    return (
      <Row justify="center">
        <ButtonMore color="default" loading={loading} onClick={this.handleClick}>Mais...</ButtonMore>
      </Row>
    )
  }

}

export default Pagination
