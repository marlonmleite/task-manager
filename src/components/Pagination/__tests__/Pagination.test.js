import React from 'react'
import { shallow } from 'enzyme'
import { Row } from 'components/Layout'
import Pagination from '../'
import ButtonMore from '../styled'

describe('Pagination', () => {
  it('should render', () => {
    const pag = shallow(<Pagination page={1} onChange={jest.fn()} hasMore />)

    expect(pag).toMatchSnapshot()
  })

  it('should not have more pages', () => {
    const pag = shallow(<Pagination page={1} onChange={jest.fn()} hasMore={false} />)
    const hasContent = pag.find(Row).exists()

    expect(hasContent).toBeFalsy()
  })

  it('should handle onClick', () => {
    const onChange = jest.fn()
    const pag = shallow(<Pagination page={1} onChange={onChange} hasMore />)

    pag.find(ButtonMore).simulate('click')

    expect(onChange).toHaveBeenCalledWith(2)
  })

  it('should not handle onClick when loading', () => {
    const onChange = jest.fn()
    const pag = shallow(<Pagination page={1} onChange={onChange} loading hasMore />)

    pag.find(ButtonMore).simulate('click')

    expect(onChange).not.toHaveBeenCalled()
  })
})
