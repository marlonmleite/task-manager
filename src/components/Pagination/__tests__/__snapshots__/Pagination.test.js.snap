// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render 1`] = `
<styled.div
  justify="center"
>
  <Styled(Button)
    color="default"
    loading={false}
    onClick={[Function]}
  >
    Mais...
  </Styled(Button)>
</styled.div>
`;
